homework/index.js

```diff
@@ -29,10 +29,68 @@
     return elem;
   }
 
-  function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+  function changeDateFormat(date) {
+    const timeFormat = new Date(date);
+    return timeFormat.toLocaleString();
+  }
+
+  function sortRepos(a, b) {
+    const first = a.name.toLowerCase();
+    const second = b.name.toLowerCase();
+    const key = first < second ? -1 : first > second ? 1 : 0;
+    return key;
   }
```

Comment:
On this sort function, I'm giving you the same feedback as I gave Esra:

In the assignment we gave the hint to use `.localeCompare`. Your method works fine if all names just use the regular letters. However, as soon as they include letters with diacritics, such as in French, German, Dutch or Turkish the names are not sorted in the (dictionary) order you would expect.

You may want to try this code, comparing your sort method agains `.localeCompare()`. I have taken the surnames of the students in class23 but added another name, viz. **Öbul** which I would expect to be sorted just after **Obul**. With `.localeCompare()` (which takes into account the 'locale' that is set up in your computer) this works fine. Your sort method however places **Öbul** after **Zengin**.

```js
'use strict';

{
  const users = [
    'Hüsrev',
    'Raad',
    'Kanatsiz',
    'Ekinci',
    'Ince',
    'Ajaj',
    'Obul',
    'Öbul',
    'Zengin',
    'Karatas',
    'Ersaru',
    'Tercan',
    'Alnakib',
    'Grabetskaia',
    'Sönmez',
    'Akkuz',
  ];

  const esraSorted = users.slice().sort((a, b) => {
    const aUpper = a.toUpperCase();
    const bUpper = b.toUpperCase();
    if (aUpper < bUpper) {
      return -1;
    }
    if (aUpper > bUpper) {
      return 1;
    }
    return 0;
  });

  const localeSorted = users.slice().sort((a, b) => a.localeCompare(b));

  console.log(esraSorted);
  console.log(localeSorted);
}
```

Reply:
I've actually seen your feedback on this. Seems, I've missed that while being busy with something else.